File: 0011-container-with-most-water/0011-container-with-most-water.c
int maxArea(int* height, int Size) {
    int i=0,max=0,j=Size-1;
    while(j>i){
        int t=height[i];
        if(t>height[j]) t=height[j];
        t*=(j-i);
        if(max<t) max=t;
        if(height[i]<height[j]) i++;
        else j--;
    }
    return max;
}

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int  max=0,n=height.length;
        int i=0,j=n-1,t=0;
        while(j>i){
            t=Math.min(height[j],height[i])*(j-i);
            max=Math.max(max,t);
            if(height[i]<height[j]) ++i;
            else --j;
        }
        return max;
    }
}